### PR TITLE
allow "internal" wearables even with skin-contact gear on top

### PR DESCRIPTION
**Pull request by misterprimus**
Wear ear plugs while wearing power armor
#### Summary

```SUMMARY: Bugfixes "Adds ability to wear under-the-skin gear like ear plugs even with skin contact-demanding items like power armor."```
#### Purpose of change

Make the game make sense by allowing one to wear earplugs underneath power armor.
#### Describe the solution

Just wear earplugs while wearing power armor. Before this, it was impossible.
#### Testing

Wore ear plugs under power armor. Other gear still cannot be worn over it.

### CODE REVIEW

**Comment by misterprimus** [comment]
> This should be done with a flag, not with a check for how many body parts it covers.

Could you explain? Isn't the "flag" the quantity of body parts being covered equaling 0?

**Comment by anothersimulacrum** [comment]
That's really unclear, and not visible to JSON contributors anywhere. Flags are both documented and clear to JSON contributors.

**Comment by misterprimus** [comment]
> That's really unclear, and not visible to JSON contributors anywhere. Flags are both documented and clear to JSON contributors.

Again, I don't follow. This automatically handles all items don't cover any body parts. My understanding is power armor requires skin contact and therefore can only be worn over a certain body part when there is nothing covering it. Thus any item in existence now or created in the future would properly be compatible with power armor if it has no body parts coverage. What flag would you suggest and how would it not be redundant?

**Comment by anothersimulacrum** [comment]
I suggest you add a new flag.

> This automatically handles all items don't cover any body parts... Thus any item in existence now or created in the future would properly be compatible with power armor if it has no body parts coverage

This is not necessarily a good thing. A flag allows for selection, which is important here.

**Comment by misterprimus** [comment]
> That's really unclear, and not visible to JSON contributors anywhere. Flags are both documented and clear to JSON contributors.

Do you mean like putting an "under_the_skin" flag to an item like "mouthpiece" which technically covers "mouth" but really should be usable under power armor helmet?

**Comment by anothersimulacrum** [comment]
Yes, that could be one use case.

**Comment by Rail-Runner** [comment]
Maybe just generalize it to "POWERAMOR_COMPATIBLE" flag? Not just under-the-skin things could be worn with power armor, I think; things like briefcases and shopping bags could be too.

**Comment by anothersimulacrum** [comment]
You need to document this flag.

File: src/player.cpp [hunk 1]
```diff
@@ -4043,7 +4043,8 @@ ret_val<bool> player::can_wear( const item &it ) const
 
     if( it.is_power_armor() ) {
         for( auto &elem : worn ) {
-            if( ( elem.get_covered_body_parts() & it.get_covered_body_parts() ).any() ) {
+            if( ( elem.get_covered_body_parts() & it.get_covered_body_parts() ).any() &
```
Comment (BevapDin) [inline src/player.cpp]:
```suggestion
            if( ( elem.get_covered_body_parts() & it.get_covered_body_parts() ).any() &&
```

File: src/player.cpp [hunk 2]
```diff
@@ -4072,8 +4073,8 @@ ret_val<bool> player::can_wear( const item &it ) const
         // Only headgear can be worn with power armor, except other power armor components.
         // You can't wear headgear if power armor helmet is already sitting on your head.
         bool has_helmet = false;
-        if( is_wearing_power_armor( &has_helmet ) &&
-            ( has_helmet || !( it.covers( bp_head ) || it.covers( bp_mouth ) || it.covers( bp_eyes ) ) ) ) {
+        if( !( it.has_flag( "POWERARMOR_COMPATIBLE" ) ) && ( ( is_wearing_power_armor( &has_helmet ) &&
```
Comment (BevapDin) [inline src/player.cpp]:
```suggestion
        if( !it.has_flag( "POWERARMOR_COMPATIBLE" ) && ( ( is_wearing_power_armor( &has_helmet ) &&
```

**Comment by misterprimus** [comment]
Getting failure:
`data/json/itemgroups/SUS/clothes_store.json needs to be linted`
but I didn't change it.

**Pull request merged**